### PR TITLE
Add symbols for adventure log related functions/structures

### PR DESCRIPTION
**Pull request by irdkwia**
Adds the adventure log structure pointer and related functions to access/write data in it.

You may want to conserve the old function names that were already defined, namely:
- ScriptSpecialProcess0x3A -> IncrementNbDungeonsCleared
- ScriptSpecialProcess0x3B -> IncrementNbBigTreasureWins
- SendSkyGiftToGuildmaster -> IncrementNbSkyGiftsSent

### CODE REVIEW

File: symbols/arm9.yml [hunk 1]
```diff
@@ -1279,30 +1279,381 @@ arm9:
...
+    - name: SetPokemonBattled
+      address:
+        NA: 0x204FE58
+        EU: 0x2050190
+        JP: 0x20501A4
+      description: |-
+        Marks one pokémon as battled.
+        
+        r0: monster ID
+    - name: GetNbPkmnBattled
```
Comment (UsernameFodder) [inline symbols/arm9.yml:1487]: For consistency with `GetNbPokemonJoined`, let's spell out `GetNbPokemonBattled` fully here too (and actually you mention "GetNbPokemonBattled" in a later description anyway).

File: symbols/arm9.yml [hunk 2]
```diff
@@ -1279,30 +1279,381 @@ arm9:
...
+        NA: 0x204FFEC
+        EU: 0x2050324
+        JP: 0x2050338
+      description: |-
+        Computes the counters from the bit fields in the adventure log, as they are not updated automatically when bit fields are altered.
+        
+        Affects GetNbPokemonJoined, GetNbMovesLearned, GetNbPokemonBattled and GetNbItemAcquired.
+        
+        No params.
+    - name: RecruitSpecialPokémonLog
```
Comment (UsernameFodder) [inline symbols/arm9.yml:1583]: Can we change "Pokémon" to "Pokemon" (without the accent) for ease of typing?

File: headers/types/common/common.h [hunk 3]
```diff
@@ -302,6 +302,40 @@ struct move_data_table {
...
+    uint32_t nb_big_treasure_wins;
+    uint32_t nb_recycled;
+    uint32_t nb_gifts_sent;
+    uint32_t nb_big_treasure_wins;
+    uint32_t pokemon_joined_flags[37];
+    uint32_t pokemon_battled_flags[37];
+    uint32_t moves_learned_flags[17];
+    uint32_t items_acquired_flags[44];
+    uint32_t special_challenge_flags;
+    uint32_t footprints_game_points[5];
```
Comment (UsernameFodder) [inline headers/types/common/common.h:334]: Does "footprints" refer to the Sentry Duty minigame? Can we refer to this as `sentry_duty_game_points` so we're sticking with the official terminology? Similarly for the `GetFootprintsGamePoints` and `SetFootprintsGamePoints` functions.

**Review by UsernameFodder — CHANGES_REQUESTED** [review]
FYI, as a first-time contributor I'll have to approve the CI runs manually. It should be easier in subsequent PRs.

Although, I'm not really why but it seems like GitHub Actions isn't running at all here for some reason.

**Comment by UsernameFodder** [comment]
> You may want to conserve the old function names that were already defined, namely:
> 
>     * ScriptSpecialProcess0x3A -> IncrementNbDungeonsCleared
> 
>     * ScriptSpecialProcess0x3B -> IncrementNbBigTreasureWins
> 
>     * SendSkyGiftToGuildmaster -> IncrementNbSkyGiftsSent

I'm not too concerned about this, since I doubt anyone is relying on these functions. Anyway, there's already been some discussion about stability of symbol names, and it should be okay to change them as needed (compatibility will be handled in other ways).

File: headers/types/common/common.h [hunk 4]
```diff
@@ -302,6 +302,40 @@ struct move_data_table {
...
+    struct dungeon_floor_pair current_floor;
```
[the remark this answers is not in the document]
Comment (irdkwia) [reply headers/types/common/common.h:334]: I added padding bytes at the end of the struct.

File: headers/functions/arm9.h [hunk 5]
```diff
@@ -135,9 +135,46 @@ int GetSpecialEpisodeType(void);
...
+void IncrementNbBigTreasureWins(void);
+void SetNbBigTreasureWins(uint32_t nb_big_treasure_wins);
+uint32_t GetNbBigTreasureWins(void);
+void SetNbRecycled(uint32_t nb_recycled);
+uint32_t GetNbRecycled(void);
+void IncrementNbSkyGiftsSent(void);
+void SetNbSkyGiftsSent(uint32_t nb_sky_gifts_sent);
+uint32_t GetNbSkyGiftsSent(void);
+void ComputeSpecialCounters(void);
+void RecruitSpecialPokémonLog(enum monster_id monster_id);
```
Comment (UsernameFodder) [inline headers/functions/arm9.h:170]: Don't forget to update the function names here as well:

```
./symbol_check.py
arm9: found 4 discrepancies between C headers and symbol tables.
The following functions are present in ../headers/functions/arm9.h but missing from ../symbols/arm9.yml:
    - RecruitSpecialPokémonLog
    - GetFootprintsGamePoints
    - SetFootprintsGamePoints
    - GetNbPkmnBattled
```

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

File: headers/types/common/common.h [hunk 6]
```diff
@@ -302,6 +302,40 @@ struct move_data_table {
...
+ASSERT_SIZE(struct dungeon_floor_pair, 2);
+
+// The adventure log structure.
+struct adventure_log {
+    uint32_t completion_flags[4];
+    uint32_t nb_dungeons_cleared;
+    uint32_t nb_friend_rescues;
+    uint32_t nb_evolutions;
+    uint32_t nb_eggs_hatched;
+    uint32_t successful_steals; //Unused in Sky
```
Comment (UsernameFodder) [inline headers/types/common/common.h:319]: The formatter wants a space in the comment: `// Unused in Sky`

**Review by UsernameFodder — CHANGES_REQUESTED** [review]

**Review by UsernameFodder — APPROVED** [review]
Alright, this all looks good to me, thanks!

I need to figure out why the tests weren't running on this PR (especially if this keeps happening)... But you can see what they were supposed to look like here: https://github.com/UsernameFodder/pmdsky-debug/pull/35/checks.

**Pull request merged**